Sources/Plasma/NucleusLib/pnAsyncCore/Private/pnAcIo.h

```diff
@@ -132,20 +132,24 @@ struct AsyncNotifySocketListen : AsyncNotifySocketConnect {
 
 struct AsyncNotifySocketRead : AsyncNotifySocket {
     uint8_t *       buffer;
-    unsigned        bytes;
-    unsigned        bytesProcessed;
+    size_t          bytes;
+    size_t          bytesProcessed;
 
     AsyncNotifySocketRead() : buffer(), bytes(), bytesProcessed() { }
 };
 
-typedef AsyncNotifySocketRead AsyncNotifySocketWrite;
 
-typedef bool (* FAsyncNotifySocketProc) (    // return false to disconnect
-    AsyncSocket         sock,
-    EAsyncNotifySocket  code,
-    AsyncNotifySocket * notify,
-    void **             userState
-);
+struct AsyncNotifySocketWrite : AsyncNotifySocketRead {
+    size_t          bytesCommitted;
+    
+    AsyncNotifySocketWrite() : AsyncNotifySocketRead(), bytesCommitted() { }
+};
+
+// return false to disconnect
+typedef std::function<bool (AsyncSocket /* sock */,
+                            EAsyncNotifySocket /* code */,
+                            AsyncNotifySocket* /* notify */,
+                            void** /* userState */)> FAsyncNotifySocketProc;
```

Comment:
```suggestion
using FAsyncNotifySocketProc = std::function<bool(AsyncSocket, EAsyncNotifySocket, AsyncNotifySocket*, void**)> ;
```

The arugment names should probably go into a doxygen comment, IMO.